resources/js/chat/chat-state-store.ts

```diff
@@ -121,50 +187,27 @@ export default class ChatStateStore implements DispatchListener {
     }
   }
 
-  // TODO: This will support the other types of joining channels in the future
-  // Only up to one join/create channel operation should be allowed to be running at any time.
-  // For consistency, the operation is considered complete when the websocket message arrives, not when the request completes.
-  @action
-  joinChannel() {
-    if (!this.createAnnouncement.isValid || this.isJoiningChannel) return;
-
-    const json = this.createAnnouncement.toJson();
-    this.waitJoinChannelUuid = json.uuid;
-    // TODO: when adding more channel types to join, remember to add separate busy spinner states for them.
-
-    createAnnouncement(json)
-      .fail(action((xhr: JQueryXHR) => {
-        onError(xhr);
-        this.waitJoinChannelUuid = null;
-      }));
-  }
-
   @action
   selectChannel(channelId: ChannelId, mode: 'advanceHistory' | 'replaceHistory' | null = 'advanceHistory') {
-    // TODO: enforce location url even if channel doesn't change;
-    // noticeable when navigating via ?sendto= on existing channel.
-    if (this.selected === channelId) return;
-
     // Mark the channel being switched away from as read.
     // Marking as read is done here to avoid constantly sending mark-as-read requests
     // while receiving messages when autoScroll is enabled on the channel.
     this.selectedChannel?.throttledSendMarkAsRead();
-
     this.selected = channelId;
 
-    if (channelId === 'create') {
+    if (channelId == null) return;
+
+    if (typeof channelId === 'string') {
       if (mode != null) {
         Turbolinks.controller[mode](updateQueryString(null, {
           channel_id: null,
           sendto: null,
-        }, 'create'));
+        }, channelId));
       }
 
       return;
     }
 
-    if (channelId == null) return;
-
     const channel = this.channelStore.get(channelId);
 
     if (channel == null) {
```

Comment:
feels like this should reset `this.selected` as well or maybe even have `this.selected` as `Channel|AddChannelType|null` if the channel models don't change.

Reply:
This shouldn't happen (since the selection comes from the list of channels), unless some funny business is being done, so it's better not to mask it, and not have `this.selected` changing to a value that wasn't requested.

Reply:
so, throw exception instead of ignoring it?

Reply:
It doesn't throw so chat doesn't become unusable if something triggers a race condition for the user.

Reply:
🤔 so it's expected to receive an invalid channel id? and then assign it to then re-check it whenever it's used?